### ISSUE TITLE
pip install error on Linux

### ISSUE THREAD

**Issue by FranzEricSchneider** [parlance/ctcdecode#209]
Hi, on Ubuntu 20.04 I am unable to install ctcdecode with pip. I get the same failures with `pip install .` as instructed in the README or `pip install git+https://github.com/parlance/ctcdecode` as suggested in a previous install issue.

The torch, gcc, and g++ are ahead of what is mentioned here: https://github.com/parlance/ctcdecode/issues/182

The error message is truly large and very hard to read, but the beginning and ending look like so:
```
$ pip install git+https://github.com/parlance/ctcdecode
Defaulting to user installation because normal site-packages is not writeable
Collecting git+https://github.com/parlance/ctcdecode
  Cloning https://github.com/parlance/ctcdecode to /tmp/pip-req-build-j7if84vf
  Running command git clone --filter=blob:none --quiet https://github.com/parlance/ctcdecode /tmp/pip-req-build-j7if84vf
  Resolved https://github.com/parlance/ctcdecode to commit c90ad94a0b19554f80804fb7812f2a1447a34a70
  Running command git submodule update --init --recursive -q
  Preparing metadata (setup.py) ... done
Building wheels for collected packages: ctcdecode
  Building wheel for ctcdecode (setup.py) ... error    
  error: subprocess-exited-with-error
  
  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [2682 lines of output]
      /home/eric/.local/lib/python3.8/site-packages/setuptools/_distutils/extension.py:134: UserWarning: Unknown Extension options: 'package', 'with_cuda'
        warnings.warn(msg)
      running bdist_wheel
      /home/eric/.local/lib/python3.8/site-packages/torch/utils/cpp_extension.py:411: UserWarning: Attempted to use ninja as the BuildExtension backend but we could not find ninja.. Falling back to using the slow distutils backend.
        warnings.warn(msg.format('we could not find ninja.'))

...

      /tmp/pip-req-build-j7if84vf/third_party/openfst-1.6.7/src/include/fst/float-weight.h: In member function ‘fst::FromGallicMapper<A, G>::ToArc fst::FromGallicMapper<A, G>::operator()(const FromArc&) const [with A = fst::ArcTpl<fst::TropicalWeightTpl<float> >; fst::GallicType G = fst::GALLIC_RESTRICT]’:
      /tmp/pip-req-build-j7if84vf/third_party/openfst-1.6.7/src/include/fst/float-weight.h:48:73: warning: ‘weight.fst::FloatWeightTpl<float>::value_’ may be used uninitialized in this function [-Wmaybe-uninitialized]
         48 |   FloatWeightTpl(const FloatWeightTpl<T> &weight) : value_(weight.value_) {}
            |                                                                         ^
      In file included from /tmp/pip-req-build-j7if84vf/third_party/openfst-1.6.7/src/include/fst/fstlib.h:39,
                       from ctcdecode/src/path_trie.h:10,
                       from ctcdecode/src/scorer.h:14,
                       from ctcdecode/src/scorer.cpp:1:
      /tmp/pip-req-build-j7if84vf/third_party/openfst-1.6.7/src/include/fst/arc-map.h:881:8: note: ‘weight.fst::FloatWeightTpl<float>::value_’ was declared here
        881 |     AW weight;
            |        ^~~~~~
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> ctcdecode

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.
```

The full error output is attached:
[error.txt](https://github.com/parlance/ctcdecode/files/9913599/error.txt)

**Comment by FranzEricSchneider** [comment]
Down in the weeds of the error file I found "`fatal error: Python.h: No such file or directory`", which I found odd, since I have python3 installed and have been running things with it.

It turns out I was missing "the header files and static libraries for python dev" (https://stackoverflow.com/questions/21530577/fatal-error-python-h-no-such-file-or-directory). I'm not quite enough of a pythoner to know what python dev vs non-dev is, but installing `python3-dev` worked to get `ctcdecode` built.

Long story short, `sudo apt-get install python3-dev` fixed the issue, closing!

**Comment by zhouliping3712** [comment]
how to solve this problem
